File: app/src/main/java/com/worldventures/dreamtrips/wallet/ui/settings/help/video/holder/WalletVideoHolderDelegate.java
package com.worldventures.dreamtrips.wallet.ui.settings.help.video.holder;

import com.worldventures.core.ui.view.custom.PinProgressButton;
import com.worldventures.dreamtrips.wallet.ui.settings.help.video.delegate.WalletVideoCallback;
import com.worldventures.dreamtrips.wallet.ui.settings.help.video.model.WalletVideoModel;

public abstract class WalletVideoHolderDelegate {

   public abstract void sendPlayVideoAction(WalletVideoModel video);

   public abstract void sendMembershipVideoAction(WalletVideoModel video);

   protected abstract WalletVideoHolderHelper createHelper(PinProgressButton progressView);

   protected interface WalletVideoHolderHelper {

      void onDownloadClick(WalletVideoCallback videoCallback, WalletVideoModel walletVideoModel);

      void setModelObject(WalletVideoModel walletVideoModel);

      void syncUIStateWithModel();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/ui/settings/help/video/holder/WalletVideoHolderDelegate.kt
package com.worldventures.wallet.ui.settings.help.video.holder

import com.worldventures.core.ui.view.custom.PinProgressButton
import com.worldventures.wallet.ui.settings.help.video.delegate.WalletVideoCallback
import com.worldventures.wallet.ui.settings.help.video.model.WalletVideoModel

abstract class WalletVideoHolderDelegate {

   abstract fun sendPlayVideoAction(video: WalletVideoModel)

   abstract fun sendMembershipVideoAction(video: WalletVideoModel)

   abstract fun createHelper(progressView: PinProgressButton): WalletVideoHolderHelper

   interface WalletVideoHolderHelper {

      fun onDownloadClick(videoCallback: WalletVideoCallback, walletVideoModel: WalletVideoModel)

      fun setModelObject(walletVideoModel: WalletVideoModel)

      fun syncUIStateWithModel()
   }
}
